<launch>
  <arg name="global_frame" default="base_link" />
  <arg name="color_image" default="/camera/color/image_raw" />
  <arg name="color_info" default="/camera/color/camera_info" />
  <arg name="depth_image" default="/camera/depth/image_raw" />
  <arg name="depth_info" default="/camera/depth/camera_info" />
  <arg name="run_action_server" default="true" />
  <arg name="modal" default="full" />

  <node pkg="nodelet" type="nodelet" name="depth_processing_nodelet"  args="manager" output="screen">
    <param name="run_action_server" type="bool" value="$(arg run_action_server)" />
    <param name="global_frame" value="$(arg global_frame)" />
    <param name="modal" value="$(arg modal)" />
    <remap from="/camera/color/image_raw" to="$(arg color_image)"/>
    <remap from="/camera/color/camera_info" to="$(arg color_info)"/>
    <remap from="/camera/depth/image_rect_raw" to="$(arg depth_image)"/>
    <remap from="/camera/depth/camera_info" to="$(arg depth_info)"/>
  </node>
  <node pkg="nodelet" type="nodelet" name="depth_nodelet" args="load gr_depth_processing/MyNodeletClass depth_processing_nodelet" output="screen"/>

 <!--node pkg="tf2_ros" type="static_transform_publisher" name="tf_broadcaster_base_camera" args="0 0 0 0 0 0 base_link camera_link" />
 <node pkg="tf2_ros" type="static_transform_publisher" name="tf_broadcaster_base_camera2" args="0 0 0 0 0 1.575 camera_link camera_depth_optical_frame" /-->
</launch>
